pallets/avn-transaction-payment/src/tests/adjustment_fee_tests.rs

```diff
@@ -0,0 +1,582 @@
+
+                    pay_gas_and_call_remark(&sender);
+                    assert_eq!(INITIAL_SENDER_BALANCE - FIXED_FEE, Balances::free_balance(sender));
+
+                    assert!(event_emitted(&mock::RuntimeEvent::AvnTransactionPayment(
+                        crate::Event::<TestRuntime>::AdjustedTransactionFeePaid {
+                            who: sender,
+                            fee: FIXED_FEE,
+                        }
+                    )));
+                })
+            }
+
+            #[test]
+            fn and_fixed_fee_higher_than_base_fee() {
+                new_test_ext().execute_with(|| {
+                    let sender = to_acc_id(1u64);
+                    set_initial_sender_balance(&sender);
+
+                    let higher_fixed_fee = u128::from(BASE_FEE + 1);
+                    let config = AdjustmentInput::<TestRuntime> {
+                        fee_type: FeeType::FixedFee(FixedFeeConfig { fee: higher_fixed_fee }),
+                        adjustment_type: AdjustmentType::None,
+                    };
+
+                    assert_ok!(AvnTransactionPayment::set_known_sender(
+                        RuntimeOrigin::root(),
+                        sender,
+                        config,
+                    ));
+                    assert_eq!(AvnTransactionPayment::is_known_sender(sender), true);
+
+                    pay_gas_and_call_remark(&sender);
+                    assert_eq!(INITIAL_SENDER_BALANCE - u128::from(BASE_FEE), Balances::free_balance(sender));
+
+                    assert!(event_emitted(&mock::RuntimeEvent::AvnTransactionPayment(
+                        crate::Event::<TestRuntime>::AdjustedTransactionFeePaid {
+                            who: sender,
+                            fee: u128::from(BASE_FEE), // sender paid base fee
+                        }
+                    )));
+                })
+            }
+
+            #[test]
+            fn and_valid_percentage_fee() {
+                new_test_ext().execute_with(|| {
+                    let sender = to_acc_id(1u64);
+                    set_initial_sender_balance(&sender);
+
+                    let config = AdjustmentInput::<TestRuntime> {
+                        fee_type: FeeType::PercentageFee(PercentageFeeConfig {
+                            percentage: PERCENTAGE_FEE,
+                            _marker: sp_std::marker::PhantomData::<TestRuntime>,
+                        }),
+                        adjustment_type: AdjustmentType::None,
+                    };
+
+                    assert_ok!(AvnTransactionPayment::set_known_sender(
+                        RuntimeOrigin::root(),
+                        sender,
+                        config,
+                    ));
+                    assert_eq!(AvnTransactionPayment::is_known_sender(sender), true);
+
+                    pay_gas_and_call_remark(&sender);
+
+                    let percentage_value = get_percentage_fee_value(PERCENTAGE_FEE);
```

Comment:
I think we should add another test condition here, to make it very clear what is the base of the percentage. Is that `base_fee`? So, is the fee we paid, say, `(1 - discount rate) * BASE_FEE`?

Reply:
I changed the `get_percentage_fee_value` to `get_percentage_fee_paid` and now returns the value paid with the percentage given.